src/docfx/build/toc/TableOfContentsLoader.cs

```diff
@@ -37,29 +41,106 @@ internal class TableOfContentsLoader
             TableOfContentsParser parser,
             MonikerProvider monikerProvider,
             DependencyMapBuilder dependencyMapBuilder,
-            ContentValidator contentValidator)
+            ContentValidator contentValidator,
+            Config config,
+            ErrorBuilder errors,
+            DocumentProvider documentProvider)
         {
             _linkResolver = linkResolver;
             _xrefResolver = xrefResolver;
             _parser = parser;
             _monikerProvider = monikerProvider;
             _dependencyMapBuilder = dependencyMapBuilder;
             _contentValidator = contentValidator;
+            _errors = errors;
+            _documentProvider = documentProvider;
+            _joinTOCConfigs = config.JoinTOC is null
+                ? new Dictionary<string, JoinTOCConfig>()
+                : config.JoinTOC.Where(x => x.ReferenceToc != null).ToDictionary(x => PathUtility.Normalize(x.ReferenceToc!));
         }
 
         public (TableOfContentsNode node, List<Document> referencedFiles, List<Document> referencedTocs)
-            Load(ErrorBuilder errors, Document file)
+            Load(Document file)
         {
             return _cache.GetOrAdd(file.FilePath, _ =>
             {
                 var referencedFiles = new List<Document>();
                 var referencedTocs = new List<Document>();
-                var node = LoadTocFile(file, file, referencedFiles, referencedTocs, errors);
+                var node = LoadTocFile(file, file, referencedFiles, referencedTocs, _errors);
 
+                if (_joinTOCConfigs.TryGetValue(file.FilePath.Path, out var joinTOCConfig) && joinTOCConfig != null)
+                {
+                    if (joinTOCConfig.TopLevelToc != null)
+                    {
+                        node = JoinToc(node, joinTOCConfig.TopLevelToc);
+                    }
+                }
                 return (node, referencedFiles, referencedTocs);
             });
         }
 
+        private TableOfContentsNode JoinToc(TableOfContentsNode referenceToc, string topLevelTocFilePath)
+        {
+            var (topLevelToc, _, _) = Load(_documentProvider.GetDocument(FilePath.Content(new PathString(topLevelTocFilePath))));
+
+            var itemsToMatch = new Dictionary<string, SourceInfo<TableOfContentsNode>>();
+            var duplicatedName = new HashSet<string>();
+            foreach (var child in referenceToc.Items)
+            {
+                var key = child.Value.Name;
+                if (string.IsNullOrEmpty(key))
+                {
+                    _errors.Add(Errors.JsonSchema.MissingAttribute(key, "name"));
+                }
+                else if (!duplicatedName.Add(key!))
+                {
+                    _errors.Add(Errors.TableOfContents.DuplicateJoinTocName(key!));
+                    itemsToMatch.Remove(key!);
```

Comment:
This doesn't seem the right way to do dup detection. Consider 3 items with the same key, the 3rd one will be added.

Reply:
It can be detected, but we agreed to skip the validation anyway